cf/read_write/netcdf/netcdfread.py

```diff
@@ -206,25 +197,77 @@ def _create_data(
                 coord_ncvar=coord_ncvar,
             )
 
+            # Set the CFA write status to True when there is exactly
+            # one dask chunk
+            if data.npartitions == 1:
+                data._cfa_set_write(True)
+
+            self._cache_data_elements(data, ncvar)
+
+            return data
+
         # ------------------------------------------------------------
-        # Still here? Then create data for a CFA-netCDF variable
+        # Still here? Create data for a CFA variable
         # ------------------------------------------------------------
-        cfa_array, kwargs = self._create_cfanetcdfarray(
-            ncvar,
-            unpacked_dtype=unpacked_dtype,
-            coord_ncvar=coord_ncvar,
-        )
+        if construct is not None:
+            # Remove the aggregation attributes from the construct
+            self.implementation.del_property(
+                construct, "aggregated_dimensions", None
+            )
+            aggregated_data = self.implementation.del_property(
+                construct, "aggregated_data", None
+            )
+        else:
+            aggregated_data = None
 
-        # Return the data
-        return self._create_Data(
+        if cfa_term:
+            term, term_ncvar = tuple(cfa_term.items())[0]
+            cfa_array, kwargs = self._create_cfanetcdfarray_term(
+                ncvar, term, term_ncvar
+            )
+        else:
+            cfa_array, kwargs = self._create_cfanetcdfarray(
+                ncvar,
+                unpacked_dtype=unpacked_dtype,
+                coord_ncvar=coord_ncvar,
+            )
+
+        data = self._create_Data(
             cfa_array,
             ncvar,
             units=kwargs["units"],
             calendar=kwargs["calendar"],
         )
 
+        # Note: We don't cache elements from CFA variables
```

Comment:
I have an idea why, but please could you clarify this to confirm or otherwise my notions, not necessarily as a comment (though potentially adding a phrase or two to this one) - just for my understanding...

Reply:
More context put into comment: d0ad8cec4132299ff4e1299df33575bfc85a90cf